src/qibosoq/rfsoc_server.py

```diff
@@ -67,8 +67,16 @@ def execute_program(self, data: dict) -> dict:
             *args,
         )
 
+        asm_prog = program.asm()
         qick_logger.handlers[0].doRollover()
-        qick_logger.info(program.asm())
+        qick_logger.info(asm_prog)
+
+        num_instructions = asm_prog.count("\n") - 2
```

Comment:
This should be equivalent to:
```suggestion
        num_instructions = len(program.prog_list) - 2
```
https://github.com/openquantumhardware/qick/blob/d393103ff324265bd33d4ba3fcb25a4ef3534101/qick_lib/qick/qick_asm.py#L2382

The advantage is that you do not have to iterate a whole string to obtain this information. And in principle you don't have to iterate at all, since `list`s knows their length.

But test it before using.

Reply:
Nice, I was looking for something similar in qick but didn't find it... thanks!